state/execution.go

```diff
@@ -296,7 +296,7 @@ func ExecTx(blockCache *BlockCache, tx_ types.Tx, runCall bool, evc events.Firea
 
 	// TODO: do something with fees
 	fees := uint64(0)
-	_s := blockCache.State() // hack to access validators and event switch.
+	_s := blockCache.State() // hack to access validators and block height
```

Comment:
It's like, the BlockCache is meant to act as a cache wrapper around the state, yet we mutate the validators directly.  We need to be careful, e.g. make sure we don't introduce a cache invalidation bug.  This should be documented for posterity.